samples/parameterization/itertools_pgen.py

```diff
@@ -0,0 +1,38 @@
+from maestrowf.datastructures.core import ParameterGenerator
```

Comment:
Should we rename this one lulesh_itertools_pgen.py perhaps, or something similar that makes it more clear what it's for?

Reply:
Yeah, that's a good idea. Stand by...

Reply:
I also had the thought, should we comment these?

Reply:
funny, just commented with the same thought on one of the other examples

Reply:
Do you want to handle that once this PR gets merged?

Reply:
Sure